### PR TITLE
Add a first draft book outline

### PR DESCRIPTION
**Pull request by mattcaswell**

### CODE REVIEW

**Comment by mattcaswell** [comment]
Any feedback on this?

File: openssl-book.tex [hunk 1]
```diff
@@ -27,14 +28,259 @@
...
+    \item{Chapter: Optimisation (Advanced Topic)}
+    \begin{outline}
+      \item{Multiblock}
+      \item{Async}
+      \item{Pipelining}
+      \item{Fragment sizes}
+      \item{Read ahead}
+    \end{outline}
+  \end{outline}
+  \item{Part: Cryptography}
```
Comment (kaduk) [inline openssl-book.tex:249]: I'm a little torn if we want to put crypto before SSL/TLS.

Comment (mattcaswell) [reply openssl-book.tex:249]: It's a good question. From a strict layering perspective libssl depends on libcrypto, so in that way it makes sense for crypto to come first. However, IMO more people are using libssl than (just) libcrypto so it made more sense to me for ssl stuff to come first.

**Comment by mattcaswell** [comment]
Ping @openssl/committers - still looking for approval for this. The outline doesn't need to be perfect at this stage. It's just a starting point.

**Comment by t-j-h** [comment]
I don't see a need to place a heartbleed section in the book - that information is available elsewhere and isn't relevant to a user.

**Comment by mattcaswell** [comment]
> I don't see a need to place a heartbleed section in the book - that information is available elsewhere and isn't relevant to a user.

I think the only mention of heartbleed is in the "OpenSSL History" subsection of the "About OpenSSL" chapter. I didn't see it as something requiring a lot of attention (maybe 1 or 2 paragraphs). But if you're going to have an OpenSSL History section it seemed like a pretty significant event to mention. I can take it out if it really offends you - but that kind of sounds like rewriting history.

**Comment by t-j-h** [comment]
It's not rewriting history - that isn't hidden - it's just this isn't a book about software defects so it just seems out of place. We have many others we could include. So perhaps a section on CVEs instead in which it can be covered?

**Comment by bernd-edlinger** [comment]
This thing with the broken random generator (in the debian) a few years ago, is also somehow
noticeable?

**Comment by t-j-h** [comment]
Then perhaps the "right" way to do it is to talk about the security severity ratings and development process.
Picking out a few notable bugs I don't think belongs in this book (at least it was unexpected to me given the stated purpose of the book).

**Comment by mattcaswell** [comment]
> It's not rewriting history - that isn't hidden - it's just this isn't a book about software defects so it just seems out of place. We have many others we could include. So perhaps a section on CVEs instead in which it can be covered?

Well like I said this is in the history section. No other CVE has had such a dramatic impact on the OpenSSL team and the way we develop OpenSSL. I kind of saw the main focus being not on the bug itself but more on the impact and subsequent changes that occurred. I could change the wording to "Changes as a result of Heartbleed" or something if you prefer.

> This thing with the broken random generator (in the debian) a few years ago, is also somehow
noticeable?

I wasn't attempting to come up with a list of notable bugs. That particular issue was important, but didn't really have an impact on the team itself. Heartbleed on the other hand led to massive changes in the team membership, which ultimately led to the development of the policies and procedures that we have in place today. So the intent was to cover heartbleed as a trigger for the subsequent changes

**Comment by t-j-h** [comment]
Now that makes sense - to cover the phases of the project - the early formation and kick off (of OpenSSL itself), the lean years, FIPS140 focus, then project refresh (triggered by a major security incident) and funding of full time developers. That makes sense.

**Comment by mattcaswell** [comment]
> So the intent was to cover heartbleed as a trigger for the subsequent changes

So thinking about it - how about I change the bullet to "Changes in the team membership and processes". In the content of that section we can mention heartbleed as being the trigger, but we don't have to talk about what heartbleed itself was.

**Comment by mattcaswell** [comment]
Ok - I added a new commit to tweak the history section to have more of a focus on what was happening with the team.

**Comment by mattcaswell** [comment]
Ping? I'd really like to get this merged. I've got other content waiting, but I'd like to get this in first. As I noted above, the outline doesn't have to be perfect at this stage. We can always amend it later.

**Review by kaduk — APPROVED** [review]
Sure, go for it.
I think we've waited long enough for any objections.

**Comment by kaduk** [comment]
This is ready now, right? @mattcaswell

**Comment by t-j-h** [comment]
my comments were addressed long ago ...

**Review by t-j-h — APPROVED** [review]

**Comment by mattcaswell** [comment]
Right - thanks for the nudge. Pushed.

**Pull request closed without merging**